integration-tests/push-to-external-registry/test.sh

```diff
@@ -87,3 +87,623 @@ verify_release_contents() {
+    
+    if [ -z "$snapshot_json" ]; then
+        echo "❌ ERROR: Could not retrieve snapshot ${snapshot_name}"
+        return 1
+    fi
+    
+    # Setup Docker config for registry access using decrypted secrets
+    # Try to use pull-secret first (for read access), fallback to push credentials
+    local auth_config=$(mktemp)
+    
+    # First try the pull secret (better for verification)
+    if yq '. | select(.metadata.name | contains("pull-secret")) | .data.".dockerconfigjson"' \
+        "${SUITE_DIR}/resources/managed/secrets/managed-secrets.yaml" 2>/dev/null | \
+        base64 -d > "${auth_config}" 2>/dev/null && [ -s "${auth_config}" ]; then
+        echo "Using pull-secret credentials for verification"
+    else
+        # Fallback to push credentials
+        echo "Pull-secret not available, using push credentials"
+        if ! yq '. | select(.metadata.name | contains("push-")) | .data.".dockerconfigjson"' \
+            "${SUITE_DIR}/resources/managed/secrets/managed-secrets.yaml" | \
+            base64 -d > "${auth_config}"; then
+            echo "❌ ERROR: Failed to get push credentials"
+            rm -f "${auth_config}"
+            return 1
+        fi
+    fi
+    
+    # Verify config was created successfully
+    if [ ! -s "${auth_config}" ]; then
+        echo "❌ ERROR: Failed to create registry auth config"
+        rm -f "${auth_config}"
+        return 1
+    fi
+    
+    local component_count
+    component_count=$(jq '.spec.components | length' <<< "${snapshot_json}")
+    
+    echo "Checking ${component_count} component(s)..."
+    echo ""
+    
+    local verification_failed=0
+    
+    for ((i=0; i<component_count; i++)); do
+        local component
+        component=$(jq -c ".spec.components[$i]" <<< "${snapshot_json}")
+        
+        local component_name
+        component_name=$(jq -r '.name' <<< "${component}")
+        
+        local container_image
+        container_image=$(jq -r '.containerImage' <<< "${component}")
+        
+        echo "Component: ${component_name}"
+        echo "  Source: ${container_image}"
+        
+        # Get source digest using oras
+        local source_digest
+        if ! source_digest=$(oras resolve --registry-config "${auth_config}" "${container_image}" 2>&1); then
+            echo "  ❌ ERROR: Failed to resolve source digest"
+            echo "     ${source_digest}"
+            verification_failed=1
+            continue
+        fi
+        
+        echo "  Source digest: ${source_digest}"
+        
+        # Try to get repository mappings from the snapshot CR
+        # Note: .repositories field only exists in trusted artifacts workspace, not in the CR
+        # This verification can only work if the snapshot CR was enriched (which is not standard)
+        local component_mapping
+        component_mapping=$(jq -c --arg comp "${component_name}" '.spec.components[] | select(.name == $comp) | .repositories[0]' <<< "${snapshot_json}")
```

Comment:
I don't think this will work. We don't modify the snapshot ever. Why would the snapshot have our repositories field in it?